source/libNsdl/src/sn_grs.c

```diff
@@ -23,7 +23,7 @@
  */
 #include <string.h>
 #include <stdlib.h>
-
+#include <assert.h>
```

Comment:
It's not used at all in any of portable the C/Nanomesh/ components. Is it getting used in the C++ bits?

We can't really assume we have a full "hosted" C99 library. But the minimum for a "freestanding" implementation is too tight, so we go a bit beyond that. So we take memcpy() and sprintf() from "hosted", but not fprintf().  I think assert() falls into the slightly-more-than-we-can assume fprintf() area.

My main concern is - does it actually do what you want on any given embedded platform, and do you end up accidentally pulling in a whole chunk of code to give you stderr for its output?

If you really want it, we can certainly create an NS_ASSERT in ns_types.h - it would map to MBED_ASSERT in mbedOS.